controllers/chat/initChatFromPost.js

```diff
@@ -9,6 +9,41 @@ const BetterSocialConstantListUtils = require('../../services/bettersocial/const
 const BetterSocialCore = require('../../services/bettersocial');
 const {generate_channel_id_for_anon_chat} = require('../../services/bettersocial/chat/allAnonChat');
 
+const get_anon_info_from_comment_post = async (client, userModel, source_id) => {
+  let comment_data = null;
+  let post_detail = await Getstream.feed.getPlainFeedById(source_id, {
+    withOwnReactions: true
+  });
+  let comments = post_detail?.own_reactions?.comment;
+  // iterate comments
+  for (const element of comments) {
+    let comment = element;
+    if (comment.user_id === userModel.user_id) {
+      comment_data = comment;
+      break;
+    }
+  }
+  if (!comment_data) {
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    comment_data = {
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name
+    };
+  } else {
+    comment_data = {
+      anon_user_info_color_code: comment_data?.data?.anon_user_info_color_code,
+      anon_user_info_color_name: comment_data?.data?.anon_user_info_color_name,
+      anon_user_info_emoji_code: comment_data?.data?.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: comment_data?.data?.anon_user_info_emoji_name
+    };
+  }
+
+  return comment_data;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `get_anon_info_from_comment_post` is fetching all comments for a post and iterating over them to find a comment from the current user. This could be inefficient if there are many comments. Consider adding a method in your service that fetches only the required comment based on user_id, which would reduce the amount of data fetched and processed.

```diff
-  let post_detail = await Getstream.feed.getPlainFeedById(source_id, {
-    withOwnReactions: true
-  });
-  let comments = post_detail?.own_reactions?.comment;
-  // iterate comments
-  for (const element of comments) {
-    let comment = element;
-    if (comment.user_id === userModel.user_id) {
-      comment_data = comment;
-      break;
-    }
-  }
+  let comment_data = await Getstream.feed.getCommentByUserId(source_id, userModel.user_id);
```

Also, consider handling the case where `Getstream.feed.getPlainFeedById` fails or returns null. Currently, if it fails or returns null, the code will throw an error when trying to access `post_detail.own_reactions.comment`.



<!-- This is an auto-generated comment by OSS CodeRabbit -->